### PR TITLE
Add own role context provider, & remove usage of kernel PurgeClientInterface

### PR DESCRIPTION
**Pull request by andrerom**
> Reopening #41

Done on top of that:
- Setup functional testing _(on 1.13 and 2.x)_
- Find a way to remove the tag either before FOS runs compiler pass on it, OR go over its defined arguments and remove kernel one.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,25 +1,45 @@
...
 branches:
     only:
         - master
 
 before_script:
-    - travis_retry composer selfupdate
     - echo "memory_limit=-1" >> ~/.phpenv/versions/$(phpenv version-name)/etc/conf.d/travis.ini
-    - travis_retry composer install --prefer-dist --no-interaction
+    - if [ "$BEHAT_OPTS" = "" ] ; then travis_retry composer selfupdate && travis_retry composer install --prefer-dist --no-interaction ; fi
+    - if [ "$BEHAT_OPTS" != "" ] ; then ./tests/.travis/prepare_behat.sh ; fi
```
Comment (bdunogier) [inline .travis.yml]: Could we change the name to not be behat ? We are used to it, but it might confuse people out :)

**Comment by andrerom** [comment]
@alongosz Did you already try to look into the following set of issues? Getting this when tryint to run functional tests against 7.x kernel:
```
  Background:                                # vendor/ezsystems/ezpublish-kernel/eZ/Bundle/EzPublishRestBundle/Features/Content/user_content.feature:3

    Given I have "administrator" permissions # eZ\Bundle\EzPublishRestBundle\Features\Context\RestContext::usePermissionsOfRole()

      UserSession value expected, got Behat\Testwork\Call\Exception\CallErrorException (RuntimeException)


  Scenario: Removing a User Content with the Content API works # vendor/ezsystems/ezpublish-kernel/eZ/Bundle/EzPublishRestBundle/Features/Content/user_content.feature:19

    Given there is a User Content                              # eZ\Bundle\EzPublishRestBundle\Features\Context\UserContentContext::thereIsAUserContent()

    When I create a delete request for this content            # eZ\Bundle\EzPublishRestBundle\Features\Context\UserContentContext::iCreateDeleteRequestForThisContent()

    And I send the request                                     # eZ\Bundle\EzPublishRestBundle\Features\Context\RestContext::sendRequest()

    Then response status code is "204"                         # eZ\Bundle\EzPublishRestBundle\Features\Context\RestContext::assertStatusCode()

    And the User this Content referred to is deleted           # eZ\Bundle\EzPublishRestBundle\Features\Context\UserContentContext::theUserThisContentReferredToIsDeleted()

  │

  ╳  Notice: Undefined property: Behat\Testwork\Call\Exception\CallErrorException::$sessionId in vendor/ezsystems/ezpublish-kernel/eZ/Bundle/EzPublishRestBundle/Features/Context/SubContext/Authentication.php line 112

  │

  └─ @AfterScenario # eZ\Bundle\EzPublishRestBundle\Features\Context\RestContext::cleanupSession()
```

**Comment by alongosz** [comment]
@andrerom No, I haven't seen this one, must be some new side-effect from recent changes or different settings (like `EZPLATFORM_BRANCH=master` instead of `2.0` - do we remember to keep master up-to-date when merging into 2.0?).

**Comment by andrerom** [comment]
> (like EZPLATFORM_BRANCH=master instead of 2.0 - do we remember to keep master up-to-date when merging into 2.0?).

yes, merges are supposed to go from `<stable branch affected> => <via all newer stable branches> => master` now. Aka `1.13 => 2.0 => master`

**Comment by alongosz** [comment]
This one is fresh and passing, looks like the same profile and suite:
https://travis-ci.org/ezsystems/ezpublish-kernel/jobs/317129585

**Comment by andrerom** [comment]
@alongosz Yes, noticed, so extracted tests here to own PR: #44

**Pull request merged**